sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -331,6 +331,15 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSparkSession {
     checkAnswer(df.select(nullif(lit(5), lit(5))), Seq(Row(null)))
   }
 
+  test("nullifzero function") {
+    val df = Seq((0)).toDF("a")
+    checkAnswer(df.selectExpr("nullifzero(0)"), Seq(Row(null)))
+    checkAnswer(df.select(nullifzero(lit(0))), Seq(Row(null)))
+
+    checkAnswer(df.selectExpr("nullifzero(a)"), Seq(Row(null)))
```

Comment:
Let's skip this and add more tests for:
1. Different types: maps, arrays, dates and so on
2. Not-nullable column
3. Types support ordering and do not support 
4. Non-foldable column

Reply:
sounds good, this is done.